QuickFIXn/ThreadedSocketAcceptor.cs

```diff
@@ -32,18 +33,55 @@ public class ThreadedSocketAcceptor : IAcceptor
         /// <param name="settings"></param>
         /// <param name="logFactory">If null, a NullFactory will be used.</param>
         /// <param name="messageFactory">If null, a DefaultMessageFactory will be created (using settings parameters)</param>
+        [Obsolete("Use \"Microsoft.Extensions.Logging.ILoggerFactory\" instead of \"QuickFix.Logger.ILogFactory\".")]
         public ThreadedSocketAcceptor(
             IApplication application,
             IMessageStoreFactory storeFactory,
             SessionSettings settings,
             ILogFactory? logFactory = null,
             IMessageFactory? messageFactory = null)
         {
-            ILogFactory lf = logFactory ?? new NullLogFactory();
+            ILoggerFactory lf = logFactory is null
+                ? NullLoggerFactory.Instance
+                : new LogFactoryAdapter(logFactory, settings);
+            IMessageFactory mf = messageFactory ?? new DefaultMessageFactory();
+            _settings = settings;
+            _sessionFactory = new SessionFactory(application, storeFactory, lf, mf);
+            _nonSessionLog = lf.CreateLogger("QuickFix");
+
+            try
+            {
+                foreach (SessionID sessionId in settings.GetSessions())
+                {
+                    SettingsDictionary dict = settings.Get(sessionId);
+                    CreateSession(sessionId, dict);
+                }
+            }
+            catch (Exception e)
+            {
+                throw new ConfigError(e.Message, e);
+            }
+        }
```

Comment:
as in AbstractInitiator, this could fairly trivially forward to the other constructor rather than duplicating it